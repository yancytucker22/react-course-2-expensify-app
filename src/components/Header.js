import React from 'react';  
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';


// <NavLink to="/create" activeClassName="is-active">Go create</NavLink><br/>
//<NavLink to="/help" activeClassName="is-active">Go help</NavLink><br/>
//<NavLink to="/edit" activeClassName="is-active">Go edit</NavLink>

// export for testing purposes
export const Header= ({ startLogout }) => (
    <header className="header"> 
    <div className="content-container">
      <div className="header__content">
          <Link className="header__title" to="/dashboard" activeclassname="is-active">
            <h1>Expensify</h1>
          </Link>
          <button className="button button--link" onClick={startLogout}>Logout</button>
          </div>
        </div>
    </header>
  );
  

  const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
  });

  // connect to redux . undefined - need no state, 
  export default connect(undefined, mapDispatchToProps)(Header);