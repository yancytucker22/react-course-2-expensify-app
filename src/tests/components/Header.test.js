import React from 'react';
import { shallow } from 'enzyme';
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import toJSON from 'enzyme-to-json';
import Header from '../../components/Header';




// virtually render component
// react-test-renderer


test('should render header correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();  
 
//    expect(toJSON(wrapper)).toMatchSnapshot();   
//  const renderer u= new ReactShallowRenderer();
//  renderer.render(<Header />);  
//  expect(renderer.getRenderOutput()).toMatchSnapshot();
//  console.log(renderer.getRenderOutput());


});