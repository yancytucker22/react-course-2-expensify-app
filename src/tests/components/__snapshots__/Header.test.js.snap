// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <div>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      to="/"
    >
      Dashboard
    </NavLink>
    <br />
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/help"
    >
      Go help
    </NavLink>
    <br />
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/create"
    >
      Go create
    </NavLink>
    <br />
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      to="/edit"
    >
      Go edit
    </NavLink>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</header>
`;
